Nfiann-bigquery-cloud-config (#6336)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/connect-data-platform/connnect-bigquery.md

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>       # BigQuery will reject the query

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>       # If a query would bill more than a gigabyte of data, then

Author: ```suggestion

```

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

|  config | info | type | example |
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |

</DetailsToggle>

<DetailsToggle alt_header="Execution project">

By default, dbt will use the specified `project`/`database` as both:

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</DetailsToggle>

> #### Impersonate service account

Author: ```suggestion
<DetailsToggle alt_header="Impersonate service account">
```

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

> For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

Author: ```suggestion
For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</DetailsToggle>
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>       dataset: my_dataset

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>       project: abc-123

Author: ```suggestion

```

|  config | info | type | example |
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |

</DetailsToggle>

<DetailsToggle alt_header="Execution project">

By default, dbt will use the specified `project`/`database` as both:

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</DetailsToggle>

<DetailsToggle alt_header="Impersonate service account">

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</DetailsToggle>

#### Job retry deadline seconds

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

> #### Job creation timeout seconds

Author: ```suggestion
<DetailsToggle alt_header="Job creation timeout seconds">
```

> Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

Author: ```suggestion
Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</DetailsToggle>
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>       method: oauth

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>       type: bigquery

Author: ```suggestion

```

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

|  config | info | type | example |
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |

</DetailsToggle>

<DetailsToggle alt_header="Execution project">

By default, dbt will use the specified `project`/`database` as both:

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</DetailsToggle>

<DetailsToggle alt_header="Impersonate service account">

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</DetailsToggle>

> #### Job retry deadline seconds

Author: ```suggestion
<DetailsToggle alt_header="Job retry deadline seconds">
```

> Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

Author: ```suggestion
Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</DetailsToggle>
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>     dev:

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>   outputs:

Author: ```suggestion

```

|  Config | Description | Type | Example |
| ------- | ----| ----- | -------- |

> | max bytes billed | info here | integer | 100000 |

Reviewer: this looks incomplete? do we want a 'info here'  ?
![Screenshot 2024-10-25 at 16 06 39](https://github.com/user-attachments/assets/eb2ea5e3-f210-4dcd-af8d-45924e498188)

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</DetailsToggle>

<DetailsToggle alt_header="Impersonate service account">

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</DetailsToggle>

<DetailsToggle alt_header="Job retry deadline seconds">

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</DetailsToggle>

<DetailsToggle alt_header="Job creation timeout seconds">

Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</DetailsToggle>

> #### Google cloud storage bucket

Author: ```suggestion
<DetailsToggle alt_header="Google cloud storage bucket">
```

> Everything you store in Cloud Storage must be placed inside a [bucket](https://cloud.google.com/storage/docs/buckets). Buckets help you organize your data and manage access to it.

Author: ```suggestion
Everything you store in Cloud Storage must be placed inside a [bucket](https://cloud.google.com/storage/docs/buckets). Buckets help you organize your data and manage access to it.

</DetailsToggle>
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>   target: dev

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> my-profile:

Author: ```suggestion

```

</DetailsToggle>

<DetailsToggle alt_header="Impersonate service account">

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</DetailsToggle>

<DetailsToggle alt_header="Job retry deadline seconds">

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</DetailsToggle>

<DetailsToggle alt_header="Job creation timeout seconds">

Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</DetailsToggle>

<DetailsToggle alt_header="Google cloud storage bucket">

Everything you store in Cloud Storage must be placed inside a [bucket](https://cloud.google.com/storage/docs/buckets). Buckets help you organize your data and manage access to it.

</DetailsToggle>

> #### Dataproc region

Author: ```suggestion
<DetailsToggle alt_header="Dataproc region">
```

A designated location in the cloud where you can run your data processing jobs efficiently. This region must match the location of your BigQuery dataset if you want to use Dataproc with BigQuery to ensure data doesn't move across regions, which can be inefficient and costly.

> For more information on [dataproc regions](https://cloud.google.com/bigquery/docs/locations), refer to the BigQuery documentation.

Author: ```suggestion
For more information on [dataproc regions](https://cloud.google.com/bigquery/docs/locations), refer to the BigQuery documentation.

</DetailsToggle>
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> ```jinja

Author: ```suggestion

```

- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

#### Priority

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

> ```

Author: ```suggestion

```

</DetailsToggle>

<DetailsToggle alt_header="Job retry deadline seconds">

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</DetailsToggle>

<DetailsToggle alt_header="Job creation timeout seconds">

Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</DetailsToggle>

<DetailsToggle alt_header="Google cloud storage bucket">

Everything you store in Cloud Storage must be placed inside a [bucket](https://cloud.google.com/storage/docs/buckets). Buckets help you organize your data and manage access to it.

</DetailsToggle>

<DetailsToggle alt_header="Dataproc region">

A designated location in the cloud where you can run your data processing jobs efficiently. This region must match the location of your BigQuery dataset if you want to use Dataproc with BigQuery to ensure data doesn't move across regions, which can be inefficient and costly.

For more information on [dataproc regions](https://cloud.google.com/bigquery/docs/locations), refer to the BigQuery documentation.

</DetailsToggle>

> #### Dataproc cluster name

Author: ```suggestion
<DetailsToggle alt_header="Dataproc cluster name">
```

A unique label you give to your group of virtual machines to help you identify and manage your data processing tasks in the cloud. When you integrate Dataproc with BigQuery, you need to provide the cluster name so BigQuery knows which specific set of resources (the cluster) to use for running the data jobs.

> Have a look at [Dataproc's document on Create a cluster](https://cloud.google.com/dataproc/docs/guides/create-cluster) for an overview on how clusters work.

Author: ```suggestion
Have a look at [Dataproc's document on Create a cluster](https://cloud.google.com/dataproc/docs/guides/create-cluster) for an overview on how clusters work.

</DetailsToggle>
```

- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

#### Priority

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

>     +location: 'us-west2'  # Set the BigQuery dataset location

Author: ```suggestion

```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

> - Click **Account home** from the left-hand side bar menu in dbt Cloud

Author: ```suggestion
- Click your name at the bottom left-hand side bar menu in dbt Cloud
- Select **Profile** from the menu
```

- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

#### Priority

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

>   my_project:

Author: ```suggestion

```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

> In BigQuery, optional configurations let you tailor settings for tasks such as running queries, loading or exporting data, and creating tables. While not necessary for basic operations, these options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

Reviewer: great intro!

> In BigQuery, optional configurations let you tailor settings for tasks such as running queries, loading or exporting data, and creating tables. While not necessary for basic operations, these options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

Reviewer: the examples you added don't seem to match the actual optional settings
```suggestion
In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.
```

To customize your optional configurations in dbt Cloud:

> 

Author: ```suggestion

- Click **Account home** from the left-hand side bar menu in dbt Cloud
- From there, click **Project** and select your BigQuery project
```

> - Select your BigQuery project from the left-hand menu

Reviewer: where do they start from ? from the user perspective, this isn't that clear and i wouldn't know where to go? is it from account settings -> connections -> click on the bigquery? or is it account settings -> projects -> click on project -> and then dev connection -> edit?

i think we can make this much more clearer so its not so confusing?

- Click **Settings** towards the top right
- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

> The following are the optional configs you can set in dbt Cloud:

Reviewer: i suggest considering organizing these into a table with a value, info/description, type (string or integer), and example so it's easier for the user to understand what to add.

Reviewer: hey @nataliefiann , unresolving this comment as it looks like there isn't a table that lists _all_ of the configs with clear info, examples, types (is it a string, is it an integer, etc.). so for example, this is a sample table that isn't fleshed out and would need tweaking so it matches our style guide if you plan on using it

```
|  config | info | type | example | 
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |
| priority | info here | string | interactive or batch |
| location | info here | string | `US`, `EU`, etc. |
…. rest of the configs…
```

> #### Priority

Reviewer: based on the google doc - it looks like 'continuous query job' is also supported (in preview) - might be worth confirming with the adapters team if this is supported for dbt and if so, we'd prob need to add it here and in the core doc.

The following are the optional configs you can set in dbt Cloud:

> <DetailsToggle alt_header="Priority">

Author: ```suggestion
<Expandable alt_header="Priority">
```

> The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

Reviewer: based on the google doc - it looks like 'continuous query job' is also supported (in preview) - might be worth confirming with the adapters team if this is supported for dbt and if so, we'd prob need to add it here and in the core doc.

Author: Querying here: https://dbt-labs.slack.com/archives/C02HE19D51R/p1729686700234579

#### Retries

> Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

Reviewer: i think a lot of the time-related fields would benefit from an example (e.g. a value or something). the question i would ask is what would i need to add for a retry? is it a number? a string?

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Retries">

Author: ```suggestion
<Expandable alt_header="Retries">
```

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Location">

Author: ```suggestion
<Expandable alt_header="Location">
```

> The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (e.g. `EU`, `US`), or a regional location (e.g. `us-west2`).

Reviewer: ```suggestion
The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).
```

> models:

Author: ```suggestion

```

> You can specify the `location` parameter in your `dbt_project.yml` file or in your `models'` configuration. You can set the `location` to either a multi-regional location (e.g., `US`, `EU`) or a regional location such as `us-west2`.

Reviewer: oh i didnt realize this was supported in the dbt_project.yml or models config? can you share where you saw this or where it's mentioned or the source code?

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Maximum bytes build">

Author: ```suggestion
<Expandable alt_header="Maximum bytes build">
```

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

```jinja

> my-profile:

Reviewer: i think we chatted about this but did you mean to include this example here? it doesn't make sense to include a profiles.yml example in dbt cloud documentation

|  config | info | type | example |
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Execution project">

Author: ```suggestion
<Expandable alt_header="Execution project">
```

```

Example output:

```

> Database Error in model debug_table (models/debug_table.sql)

Reviewer: do we need this here?

  Query exceeded limit for bytes billed: 1000000000. 2000000000 or higher required.
  compiled SQL at target/run/bq_project/models/debug_table.sql

```

> #### Execution project

Reviewer: as mentioned [here](https://github.com/dbt-labs/docs.getdbt.com/pull/6336/files#r1812401112), this would benefit from an example or something?

Author: Example queried: https://dbt-labs.slack.com/archives/C067EUS2D7S/p1729762435076769

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

By default, dbt will use the specified `project`/`database` as both:

> 1. The location to materialize resources (models, seeds, snapshots, etc), unless they specify a custom project/database config

Reviewer: ```suggestion
1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
```

> <DetailsToggle alt_header="Impersonate service account">

Author: ```suggestion
<Expandable alt_header="Impersonate service account">
```

2. The GCP project that receives the bill for query costs or slot usage

> Optionally, you may specify an execution_project to bill for query execution, instead of the project/database where you materialize most resources.

Reviewer: ```suggestion
Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.
```

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

#### Impersonate service account

> This feature allows users authenticating via local OAuth to access BigQuery resources based on the permissions of a service account.

Reviewer: as mentioned [here](https://github.com/dbt-labs/docs.getdbt.com/pull/6336/files#r1812401112), this would benefit from an example or something?

> This feature allows users authenticating via local OAuth to access BigQuery resources based on the permissions of a service account.

Reviewer: ```suggestion
This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.
```

> <DetailsToggle alt_header="Job retry deadline seconds">

Author: ```suggestion
<Expandable alt_header="Job retry deadline seconds">
```

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

> #### Job retry deadline seconds

Reviewer: as mentioned [here](https://github.com/dbt-labs/docs.getdbt.com/pull/6336/files#r1812401112), this would benefit from an example or something?

> <DetailsToggle alt_header="Job creation timeout seconds">

Author: ```suggestion
<Expandable alt_header="Job creation timeout seconds">
```

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

> #### Job creation timeout seconds

Reviewer: as mentioned [here](https://github.com/dbt-labs/docs.getdbt.com/pull/6336/files#r1812401112), this would benefit from an example or something?

Author: Querying examples here: https://dbt-labs.slack.com/archives/C04KKUHLT32/p1729765511296929

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Google cloud storage bucket">

Author: ```suggestion
<Expandable alt_header="Google cloud storage bucket">
```

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Dataproc region">

Author: ```suggestion
<Expandable alt_header="Dataproc region">
```

A designated location in the cloud where you can run your data processing jobs efficiently. This region must match the location of your BigQuery dataset if you want to use Dataproc with BigQuery to ensure data doesn't move across regions, which can be inefficient and costly.

For more information on [dataproc regions](https://cloud.google.com/bigquery/docs/locations), refer to the BigQuery documentation.

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

> <DetailsToggle alt_header="Dataproc cluster name">

Author: ```suggestion
<Expandable alt_header="Dataproc cluster name">
```

A unique label you give to your group of virtual machines to help you identify and manage your data processing tasks in the cloud. When you integrate Dataproc with BigQuery, you need to provide the cluster name so BigQuery knows which specific set of resources (the cluster) to use for running the data jobs.

Have a look at [Dataproc's document on Create a cluster](https://cloud.google.com/dataproc/docs/guides/create-cluster) for an overview on how clusters work.

> </DetailsToggle>

Author: ```suggestion
</Expandable>
```

- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

#### Priority

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

> ```yml

Author: ```suggestion

```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

- Click **Account home** from the left-hand side bar menu in dbt Cloud
- From there, click **Project** and select your BigQuery project
- Select your BigQuery project from the left-hand menu

> - Click **Settings** towards the top right

Author: ```suggestion
- Click **Settings** towards the top right and a **Project details** window will open
```

- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

#### Priority

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

> You can specify the `location` parameter in your `dbt_project.yml` file or in your `models'` configuration. You can set the `location` to either a multi-regional location (e.g., `US`, `EU`) or a regional location such as `us-west2`.

Author: ```suggestion

```

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</Expandable>

<Expandable alt_header="Impersonate service account">

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</Expandable>

<Expandable alt_header="Job retry deadline seconds">

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</Expandable>

<Expandable alt_header="Job creation timeout seconds">

Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</Expandable>

> 

Author: ```suggestion
#### Run dbt python models on Google Cloud Platform
```

</Expandable>

<Expandable alt_header="Retries">

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

</Expandable>

<Expandable alt_header="Location">

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

</Expandable>

<Expandable alt_header="Maximum bytes billed">

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

> | Max bytes billed | The max number of bytes that can be billed for a given BigQuery query   | Integer | `1000000000` |

Author: ```suggestion

```

</Expandable>

<Expandable alt_header="Retries">

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

</Expandable>

<Expandable alt_header="Location">

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

</Expandable>

<Expandable alt_header="Maximum bytes billed">

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

> | ---------------- | ------------------------------------------------------------------------| ------- | -------- |

Author: ```suggestion

```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

> - Click your name at the bottom left-hand side bar menu in dbt Cloud

Author: ```suggestion
1. Click your name at the bottom left-hand side bar menu in dbt Cloud
```

> - Select **Your profile** from the menu

Author: ```suggestion
2. Select **Your profile** from the menu
```

> - From there, click **Projects** and select your BigQuery project

Author: ```suggestion
3. From there, click **Projects** and select your BigQuery project
```

> - Select your BigQuery project from the left-hand menu

Author: ```suggestion
4. Select your BigQuery project from the left-hand menu
```

> Go to **Development Connection** and select BigQuery

Author: ```suggestion
5. Go to **Development Connection** and select BigQuery
```

> - Scroll down to **Development Connection** and select BigQuery

Author: ```suggestion
Go to **Development Connection** and select BigQuery
```

> - Click **Edit** and then scroll down to **Optional settings**

Author: ```suggestion
6. Click **Edit** and then scroll down to **Optional settings**
```

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configurations you can set in dbt Cloud:

| Configuration                     | Information                                                                                                                  | Type    | Example                     |
|--------------------------------|------------------------------------------------------------------------------------------------------------------------------|---------|-----------------------------|
| [Priority](#priority)             | Sets the priority for BigQuery jobs (either immediate or queued for batch processing)                                       | String  | `batch` or `interactive`    |

> | Retries                        | Specifies the number of retries for failed jobs due to temporary issues                                                      | Integer | `3`                         |

Author: ```suggestion
| Retries (#retries)                       | Specifies the number of retries for failed jobs due to temporary issues                                                      | Integer | `3`                         |
```

> | Retries (#retries)                       | Specifies the number of retries for failed jobs due to temporary issues                                                      | Integer | `3`                         |

Author: ```suggestion
| [Retries](#retries)                       | Specifies the number of retries for failed jobs due to temporary issues                                                      | Integer | `3`                         |
```

> | Location                       | Location for creating new datasets                                                                                           | String  | `US`, `EU`, `us-west2`      |

Author: ```suggestion
| Location(#location)                       | Location for creating new datasets                                                                                           | String  | `US`, `EU`, `us-west2`      |
```

> | Location(#location)                       | Location for creating new datasets                                                                                           | String  | `US`, `EU`, `us-west2`      |

Author: ```suggestion
| [Location](#location)                       | Location for creating new datasets                                                                                           | String  | `US`, `EU`, `us-west2`      |
```

> | Maximum bytes billed           | Limits the maximum number of bytes that can be billed for a query                                                            | Integer | `1000000000`                |

Author: ```suggestion
| [Maximum bytes billed](#Maximum bytes billed)           | Limits the maximum number of bytes that can be billed for a query                                                            | Integer | `1000000000`                |
```

> | Impersonate service account    | Allows users authenticated locally to access BigQuery resources under a specified service account                           | String  | `service-account@project.iam.gserviceaccount.com` |

Author: ```suggestion
| [Impersonate service account](#Impersonate service account)    | Allows users authenticated locally to access BigQuery resources under a specified service account                           | String  | `service-account@project.iam.gserviceaccount.com` |
```

> | Execution project              | Specifies the project ID to bill for query execution                                                                         | String  | `my-project-id`             |

Author: ```suggestion
| [Execution project](#Execution project)              | Specifies the project ID to bill for query execution                                                                         | String  | `my-project-id`             |
```

> | Job retry deadline seconds     | Sets the total number of seconds BigQuery will attempt to retry a job if it fails                                            | Integer | `600`                       |

Author: ```suggestion
| [Job retry deadline seconds](#Job retry deadline seconds)     | Sets the total number of seconds BigQuery will attempt to retry a job if it fails                                            | Integer | `600`                       |
```

> | Job creation timeout seconds   | Specifies the maximum timeout for the job creation step                                                                      | Integer | `120`                       |

Author: ```suggestion
| [Job creation timeout seconds](#Job creation timeout seconds )   | Specifies the maximum timeout for the job creation step                                                                      | Integer | `120`                       |
```

> | Google Cloud Storage bucket    | Location for storing objects in Google Cloud Storage                                                                         | String  | `my-bucket`                 |

Author: ```suggestion
| [Google Cloud Storage bucket](#Google Cloud Storage bucket)    | Location for storing objects in Google Cloud Storage                                                                         | String  | `my-bucket`                 |
```

> | Dataproc region                | Specifies the cloud region for running data processing jobs                                                                  | String  | `US`, `EU`, `asia-northeast1` |

Author: ```suggestion
| [Dataproc region](#Dataproc region)                | Specifies the cloud region for running data processing jobs                                                                  | String  | `US`, `EU`, `asia-northeast1` |
```

> | Dataproc cluster name          | Assigns a unique identifier to a group of virtual machines in Dataproc                                                       | String  | `my-cluster`                |

Author: ```suggestion
| [Dataproc cluster name](#Dataproc cluster name)          | Assigns a unique identifier to a group of virtual machines in Dataproc                                                       | String  | `my-cluster`                |
```

</Expandable>

<Expandable alt_header="Retries">

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

</Expandable>

<Expandable alt_header="Location">

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

</Expandable>

<Expandable alt_header="Maximum bytes billed">

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

> |  Config          | Description                                                             | Type    | Example |

Author: ```suggestion

```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

> - Click your name at the bottom left-hand side bar menu in dbt Cloud

Reviewer: two things:
- since these are steps, they should be numbered [per our style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md#lists)

- these steps are a little confusing -- i don't see any 'settings' option when i click on the bq project. i think you had it right the first time using 'account home' or account name' as the first step:
1. Click **Account home** from the left-hand sidebar menu in dbt Cloud and then **Account settings**.
2. From there, click **Project** and select your BigQuery project
3. Select the **BigQuery** under **Development Connection**. 
4. Click **Edit** and then scroll down to **Optional settings**.

i feel this is more direct and shortens the steps to 4 instead of 7.

> - Select **Profile** from the menu

Author: ```suggestion
- Select **Your profile** from the menu
```

> - From there, click **Project** and select your BigQuery project

Author: ```suggestion
- From there, click **Projects** and select your BigQuery project
```

- Select your BigQuery project from the left-hand menu

> - Click **Settings** towards the top right and a **Project details** window will open

Author: ```suggestion

```

- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

> The following are the optional configs you can set in dbt Cloud:

Reviewer: ```suggestion
The following are the optional configurations you can set in dbt Cloud:
```

> | Config                         | Information                                                                                                                  | Type    | Example                     |

Reviewer: let's spell out the full word here
```suggestion
| Configuration                     | Information                                                                                                                  | Type    | Example                     |
```

|--------------------------------|------------------------------------------------------------------------------------------------------------------------------|---------|-----------------------------|

> | Priority                       | Sets the priority for BigQuery jobs (either immediate or queued for batch processing)                                       | String  | `batch` or `interactive`    |

Reviewer: since you've used expandables, do you think the ALL of the configs would benefit from linking to the expendables for more details so users know? if you agree, i woudl suggest linking to the relevant expendables for each respective option:
```suggestion
| [Priority](#priority)             | Sets the priority for BigQuery jobs (either immediate or queued for batch processing)                                       | String  | `batch` or `interactive`    |
```

> | Priority                       | Sets the priority for BigQuery jobs (either immediate or queued for batch processing)                                       | String  | `batch` or `interactive`    |

Reviewer: the information column says 'immediate or queued for batch' but the example is interactive? is immediate a typo? 

i suggest double checking the info column with each description to make sure it's all accurate.

| Retries                        | Specifies the number of retries for failed jobs due to temporary issues                                                      | Integer | `3`                         |
| Location                       | Location for creating new datasets                                                                                           | String  | `US`, `EU`, `us-west2`      |
| Maximum bytes billed           | Limits the maximum number of bytes that can be billed for a query                                                            | Integer | `1000000000`                |

> | Impersonate service account    | Allows users authenticated locally to access BigQuery resources under a specified service account                           | String  | `service-account@project.iam.gserviceaccount.com` |

Reviewer: in the expandables, execution project is after max bytes billed. should you order the table so it matches the expandable order?

| Dataproc cluster name          | Assigns a unique identifier to a group of virtual machines in Dataproc                                                       | String  | `my-cluster`                |

<Expandable alt_header="Priority">

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

</Expandable>

<Expandable alt_header="Retries">

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

</Expandable>

<Expandable alt_header="Location">

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

</Expandable>

> <Expandable alt_header="Maximum bytes build">

Reviewer: this is a typo
```suggestion
<Expandable alt_header="Maximum bytes billed">
```

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

> |  Config          | Description                                                             | Type    | Example |

Reviewer: @nataliefiann , i'm confused - why do you still need this table?

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</Expandable>

<Expandable alt_header="Impersonate service account">

This feature allows users authenticating using local OAuth to access BigQuery resources based on the permissions of a service account.

For a general overview of this process, see the official docs for [Creating Short-lived Service Account Credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct).

</Expandable>

<Expandable alt_header="Job retry deadline seconds">

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</Expandable>

<Expandable alt_header="Job creation timeout seconds">

Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</Expandable>

> 

Reviewer: gcs buckec, dataproc cluster name, and dataproc region are for running dbt python mods on gcp. there's not mention of that here in the doc so do you think we should have a mini h4 header separating the remainder of the options telling users that the following are for that? we do mention it in the core docs: https://docs.getdbt.com/docs/core/connect-data-platform/bigquery-setup#running-python-models-on-dataproc

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

1. Click your name at the bottom left-hand side bar menu in dbt Cloud
2. Select **Your profile** from the menu
3. From there, click **Projects** and select your BigQuery project
5. Go to **Development Connection** and select BigQuery
6. Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configurations you can set in dbt Cloud:

> | Configuration    | <div style={{width:'300px'}}>Information</div>   | Type    | Example            |

Reviewer: added this to fix the table width bc it was looking weird 
<img width="749" alt="Screenshot 2024-10-28 at 15 58 41" src="https://github.com/user-attachments/assets/a09e3cb2-7c12-44c0-9e91-7b950b734f7d">

|--------------------------------|------------------------------------------------------------------------------------------------------------------------------|---------|-----------------------------|

> | [Priority](#priority)             | Sets the priority for BigQuery jobs (either immediate or queued for batch processing)                                       | String  | `batch` or `interactive`    |

Author: ```suggestion
| [Priority](#priority)             | Sets the priority for BigQuery jobs (either `interactive` or queued for `batch` processing)                                       | String  | `batch` or `interactive`    |
```

</Expandable>

<Expandable alt_header="Job retry deadline seconds">

Job retry deadline seconds is the maximum amount of time BigQuery will spend retrying a job before it gives up.

</Expandable>

<Expandable alt_header="Job creation timeout seconds">

Job creation timeout seconds is the maximum time BigQuery will wait to start the job. If the job doesn’t start within that time, it times out.

</Expandable>

#### Run dbt python models on Google Cloud Platform

import BigQueryDataproc from '/snippets/_bigquery-dataproc.md';

<BigQueryDataproc />

<Expandable alt_header="Google cloud storage bucket">

Everything you store in Cloud Storage must be placed inside a [bucket](https://cloud.google.com/storage/docs/buckets). Buckets help you organize your data and manage access to it.

</Expandable>

<Expandable alt_header="Dataproc region">

A designated location in the cloud where you can run your data processing jobs efficiently. This region must match the location of your BigQuery dataset if you want to use Dataproc with BigQuery to ensure data doesn't move across regions, which can be inefficient and costly.

> For more information on [dataproc regions](https://cloud.google.com/bigquery/docs/locations), refer to the BigQuery documentation.

Reviewer: Dataproc is a product i believe so we should capitalize this
```suggestion
For more information on [Dataproc regions](https://cloud.google.com/bigquery/docs/locations), refer to the BigQuery documentation.
```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

1. Click your name at the bottom left-hand side bar menu in dbt Cloud
2. Select **Your profile** from the menu
3. From there, click **Projects** and select your BigQuery project

> 4. Select your BigQuery project from the left-hand menu

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> 

Author: ```suggestion
If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

|  config | info | type | example |
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> ```

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>   compiled SQL at target/run/bq_project/models/debug_table.sql

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

>   Query exceeded limit for bytes billed: 1000000000. 2000000000 or higher required.

Author: ```suggestion

```

To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [BigQuery-specific configuration](/reference/resource-configs/bigquery-configs).

### Optional configurations

In BigQuery, optional configurations let you tailor settings for tasks such as query priority, dataset location, job timeout, and more. These options give you greater control over how BigQuery functions behind the scenes to meet your requirements.

To customize your optional configurations in dbt Cloud:

- Click **Account home** from the left-hand side bar menu in dbt Cloud
- From there, click **Project** and select your BigQuery project
- Select your BigQuery project from the left-hand menu
- Click **Settings** towards the top right and a **Project details** window will open
- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

> #### Priority

Author: ```suggestion
<DetailsToggle alt_header="Priority">
```

> The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

Author: ```suggestion
The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

</DetailsToggle>
```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> Database Error in model debug_table (models/debug_table.sql)

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> ```

Author: ```suggestion

```

To customize your optional configurations in dbt Cloud:

- Click **Account home** from the left-hand side bar menu in dbt Cloud
- From there, click **Project** and select your BigQuery project
- Select your BigQuery project from the left-hand menu
- Click **Settings** towards the top right and a **Project details** window will open
- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

<DetailsToggle alt_header="Priority">

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

</DetailsToggle>

> #### Retries

Author: ```suggestion
<DetailsToggle alt_header="Retries">
```

> Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

Author: ```suggestion
Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

</DetailsToggle>
```

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> Example output:

Author: ```suggestion

```

#### Retries

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

#### Location

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

#### Maximum bytes build

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

> ```

Author: ```suggestion

```

- Click **Account home** from the left-hand side bar menu in dbt Cloud
- From there, click **Project** and select your BigQuery project
- Select your BigQuery project from the left-hand menu
- Click **Settings** towards the top right and a **Project details** window will open
- Scroll down to **Development Connection** and select BigQuery
- Click **Edit** and then scroll down to **Optional settings**

<Lightbox src="/img/bigquery/bigquery-optional-config.png" width="70%" title="BigQuery optional configuration"/>

The following are the optional configs you can set in dbt Cloud:

<DetailsToggle alt_header="Priority">

The `priority` for the BigQuery jobs that dbt executes can be configured with the `priority` configuration in your BigQuery profile. The priority field can be set to one of `batch` or `interactive`. For more information on query priority, consult the [BigQuery documentation](https://cloud.google.com/bigquery/docs/running-queries).

</DetailsToggle>

<DetailsToggle alt_header="Retries">

Retries in BigQuery help to ensure that jobs complete successfully by trying again after temporary failures, making your operations more robust and reliable.

</DetailsToggle>

> #### Location

Author: ```suggestion
<DetailsToggle alt_header="Location">
```

The `location` of BigQuery datasets can be set using the `location` setting in a BigQuery profile. As per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/locations), `location` may be either a multi-regional location (for example, `EU`, `US`), or a regional location (like `us-west2`).

> 

Author: ```suggestion
</DetailsToggle>
```

> #### Maximum bytes build

Author: ```suggestion
<DetailsToggle alt_header="Maximum bytes build">
```

>       maximum_bytes_billed: 1000000000

Author: ```suggestion

```

When a `maximum_bytes_billed` value is configured for a BigQuery profile, that allows you to limit how much data your query can process. It’s a safeguard to prevent your query from accidentally processing more data than you expect, which could lead to higher costs. Queries executed by dbt will fail if they exceed the configured maximum bytes threshhold. This configuration should be supplied as an integer number of bytes.

If your `maximum_bytes_billed` is 1000000000, you would enter that value in the `maximum_bytes_billed` field in dbt cloud.

|  config | info | type | example |
| ------- | ----| ----- | -------- |
| max bytes billed | info here | integer | 100000 |

> 

Author: ```suggestion
</DetailsToggle>
```

> #### Execution project

Author: ```suggestion
<DetailsToggle alt_header="Execution project">
```

By default, dbt will use the specified `project`/`database` as both:

1. The location to materialize resources (models, seeds, snapshots, and so on), unless they specify a custom project/database config
2. The GCP project that receives the bill for query costs or slot usage

> Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

Author: ```suggestion
Optionally, you may specify an execution project to bill for query execution, instead of the project/database where you materialize most resources.

</DetailsToggle>
```
